src/librustc_mir/transform/mod.rs

```diff
@@ -267,54 +269,69 @@ fn run_optimization_passes<'tcx>(
     def_id: DefId,
     promoted: Option<Promoted>,
 ) {
+    let post_borrowck_cleanup: &[&dyn MirPass<'tcx>] = &[
+        // Remove all things only needed by analysis
+        &no_landing_pads::NoLandingPads::new(tcx),
+        &simplify_branches::SimplifyBranches::new("initial"),
+        &remove_noop_landing_pads::RemoveNoopLandingPads,
+        &cleanup_post_borrowck::CleanupNonCodegenStatements,
+        &simplify::SimplifyCfg::new("early-opt"),
+        // These next passes must be executed together
+        &add_call_guards::CriticalCallEdges,
+        &elaborate_drops::ElaborateDrops,
+        &no_landing_pads::NoLandingPads::new(tcx),
+        // AddMovesForPackedDrops needs to run after drop
+        // elaboration.
+        &add_moves_for_packed_drops::AddMovesForPackedDrops,
+        // `AddRetag` needs to run after `ElaborateDrops`. Otherwise it should run fairly late,
+        // but before optimizations begin.
+        &add_retag::AddRetag,
+        &simplify::SimplifyCfg::new("elaborate-drops"),
+        // No lifetime analysis based on borrowing can be done from here on out.
+    ];
+
+    let optimizations: &[&dyn MirPass<'tcx>] = &[
+        // Optimizations begin.
+        &unreachable_prop::UnreachablePropagation,
+        &uninhabited_enum_branching::UninhabitedEnumBranching,
+        &simplify::SimplifyCfg::new("after-uninhabited-enum-branching"),
+        &inline::Inline,
+        // Lowering generator control-flow and variables
+        // has to happen before we do anything else to them.
+        &generator::StateTransform,
+        &instcombine::InstCombine,
+        &const_prop::ConstProp,
+        &simplify_branches::SimplifyBranches::new("after-const-prop"),
+        // Deaggregation isn't technically an optimization but it unlocks opportunities
+        // for later optimization passes.
+        &deaggregator::Deaggregator,
```

Comment:
So now we do skip `Deaggregator` with opt-level 0? Is that deliberate?

Reply:
I don't think it's an issue, codegen already needs to be able to handle aggregates

Reply:
Bootstrap completed to stage2 with `RUSTFLAGS="-Zmir-opt-level=0"` so it seems to be fine.

Reply:
It only needs to handle array aggregates currently. In cg_clif, that is the only aggregate kind I handle. Even the fact that array aggregates are not removed, may be changed in the future:

https://github.com/rust-lang/rust/blob/8926bb497d9b127eb318aea5aed0e745d8381591/src/librustc_mir/transform/deaggregator.rs#L17-L20

Reply:
The comment is somewhat confusing then, though.